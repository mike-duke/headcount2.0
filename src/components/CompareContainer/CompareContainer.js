import React from 'react';
import './CompareContainer.scss';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';

const CompareContainer = ({ 
  compareMessage, 
  districtsToCompare, 
  addDistrictToCompare,
  districtAverages,
  resetComparison }) => {

  const cardsToCompare = districtsToCompare.map((district) => {
    return <Card location={district.location}
                  stats={district.stats}
                  key={district.location}
                  addDistrictToCompare={addDistrictToCompare} />
  })

  let compareCard = (
    <article className="compare-card">
      {compareMessage}
    </article>
  )

  if (districtsToCompare.length >= 2) {
    let location1 = districtsToCompare[0].location;
    let location2 = districtsToCompare[1].location;
    compareCard = (
      <article className="compare-card full-card">
        <div className="location1">
          <h2>{location1}</h2>
          <p>average over time {districtAverages[location1]}</p>
        </div>
        <h1 className="yellow">compared averages</h1> 
        <p className="yellow">{districtAverages.compared}</p>
        <div className="location2">
          <h2>{location2}</h2>
          <p>average over time {districtAverages[location2]}</p>
        </div>
      </article>
    )
  }

  return (
    <section className="compare-container card-container">
      {cardsToCompare[0]}
      <div className="comparison-area">
        {compareCard}
        <button className="reset-button" onClick={resetComparison}>Reset Comparison</button>
      </div>
      {cardsToCompare[1]}
    </section>
  )
}

CompareContainer.propTypes = {
  compareMessage: PropTypes.string.isRequired,
  districtsToCompare: PropTypes.array.isRequired,
  addDistrictToCompare: PropTypes.func.isRequired,
  districtAverages: PropTypes.object.isRequired,
  resetComparison: PropTypes.func.isRequired
}

export default CompareContainer;