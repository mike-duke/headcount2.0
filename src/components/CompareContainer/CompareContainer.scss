.compare-container {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

.compare-card {
  //import Card.scss and extend card class
  border: 2px solid black;
  width: 300px;
  height: 100px;
  background: gray;
  color: white;
  margin: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  transition: all 0.3s ease;
  overflow: hidden;
  border-radius: 20px;

  h2, h1, p {
    margin: 0;
  }

  .yellow {
    color: yellow;
  }

  .location1 {
    margin-bottom: 25px;
    text-align: left;
  }

  .location2 {
    margin-top: 25px;
    text-align: right;
  }
}

.full-card {
  height: 200px;
}

.reset-button {
  all: unset;
  background: purple;
  color: white;
  border: 2px solid black;
  border-radius: 20px;
  cursor: pointer;
  width: 200px;
  height: 50px;
}