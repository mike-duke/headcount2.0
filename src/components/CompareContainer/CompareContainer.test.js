import React from 'react';
import { shallow } from 'enzyme';
import CompareContainer from './CompareContainer.js';

describe('CompareContainer', () => {

  it('should match the snapshot with no selected cards', () => {
    const wrapper = shallow(<CompareContainer compareMessage="this is only a test"
                                              districtsToCompare={[]}
                                              addDistrictToCompare={jest.fn()}
                                              districtAverages={{}}
                                              resetComparison={jest.fn} />);

    expect(wrapper).toMatchSnapshot();
  })
})