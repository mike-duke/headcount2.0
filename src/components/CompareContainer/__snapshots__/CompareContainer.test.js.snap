// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareContainer should match the snapshot with no selected cards 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    addDistrictToCompare={[MockFunction]}
    compareMessage="this is only a test"
    districtAverages={Object {}}
    districtsToCompare={Array []}
    resetComparison={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="comparison-area"
        >
          <article
            className="compare-card"
          >
            this is only a test
          </article>
          <button
            className="reset-button"
            onClick={[Function]}
          >
            Reset Comparison
          </button>
        </div>,
        undefined,
      ],
      "className": "compare-container card-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="compare-card"
            >
              this is only a test
            </article>,
            <button
              className="reset-button"
              onClick={[Function]}
            >
              Reset Comparison
            </button>,
          ],
          "className": "comparison-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this is only a test",
              "className": "compare-card",
            },
            "ref": null,
            "rendered": "this is only a test",
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset Comparison",
              "className": "reset-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset Comparison",
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="comparison-area"
          >
            <article
              className="compare-card"
            >
              this is only a test
            </article>
            <button
              className="reset-button"
              onClick={[Function]}
            >
              Reset Comparison
            </button>
          </div>,
          undefined,
        ],
        "className": "compare-container card-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="compare-card"
              >
                this is only a test
              </article>,
              <button
                className="reset-button"
                onClick={[Function]}
              >
                Reset Comparison
              </button>,
            ],
            "className": "comparison-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this is only a test",
                "className": "compare-card",
              },
              "ref": null,
              "rendered": "this is only a test",
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset Comparison",
                "className": "reset-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset Comparison",
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
